File: microservices/messageservice/src/main/java/com/example/messageservice/Service/MessageService.java
package com.example.messageservice.Service;

import com.example.messageservice.Model.Message;

import java.util.List;
import java.util.Map;
import java.util.Optional;

public interface MessageService {
    Optional<Message> get(String id);
    List<Message> findAll();
    Message save(Message m);
    void delete(String id);
    List<Message> getChat(String user1, String user2);
}

File: frontend/dislinkt-xws-front/src/app/services/message.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Message } from '../model/message';

@Injectable({
  providedIn: 'root'
})
export class MessageService {

  constructor(private http: HttpClient) { }

  public getChat(user1: string, user2: string) : Observable<Message[]> {
    return this.http.get<Message[]>('api/messages/chat/' + user1 + '/' + user2);
  }
}
